### ISSUE TITLE
Yaml dump not considering \u00a0

### ISSUE THREAD

**Issue by Lozik** [symfony/symfony#34867]
Hello everyone. Here is a 'bug' that I noticed. Or is has that a special reason why it is not considered?

**Symfony version(s) affected**: 4.2 (seems to still be the case in 4.4 and 5.0)

**Description**  
If one dumps an array using the `Yaml::dump()` function and the string value of one key in the array contains the escape character `\u00a0` (non breaking space), the string will be escaped only by single quotes (`'`), rather than double quotes (`"`). This makes it impossible to be parsed as UTF-8 character afterwards.

**How to reproduce**  
This code tries to put a non breaking space before every colon in a yaml language file (instead of a normal space).
```php
// $lang = Yaml::parseFile(...);
foreach($lang as $k => &$v) {
    $v = str_replace(' :', "\u00a0:", $v);
}
exit(Yaml::dump($lang, 20);
```

**Possible Solution**  
I've noticed that in the Dumper, `\n` is considered and thus, double quotes are used. However for `\u00a0` this is not the case. But it would be possible to consider that the same way than `\n` is.

**Additional context**  
Usage/Need: In French e.g. one needs to put a space before every colon in texts. The same holds for question marks, exclamation points and besides quotes. To not have the browser break those up, one would need to use non breaking spaces. IMHO, non breaking spaces should be compatible.

**Edit**  
After additional research, I found that `\xc2\xa0` seems to be 'the same thing' as `\u00a0` and that one works. Is there any reason why that has been included instead?

**Comment by xabbuh** [comment]
I am actually not sure if I understand the issue. Did you expect `\u00a0` to be interpreted by PHP inside a string literal?

**Comment by Lozik** [comment]
Well, I was expecting the Yaml dumper to return the strings including the `\u00a0` enclosed with double quotes, not single ones in order to be interpretable by PHP as non breaking space. Another possibility would be to replace each `\u00a0` in the dumper by a `\_` in the Yaml dump (but again enclosed with double quotes). So i.e. `\u00a0` should be seen as synonym of `\xc2\xa0`. I only found out that it was correctly dumping the latter one looking at tests.

So, this code
```php
$a = array(
    'p' => "Prénom\xc2\xa0:",
    'n' => "Nom\u00a0:"
);
echo Yaml::dump($a);
```
currently produces the following output
```yaml
'p': "Prénom\_:"
'n': 'Nom\u00a0:'
```

but I would expect something like this
```yaml
'p': "Prénom\_:"
'n': "Nom\u00a0:"
```

or this
```yaml
'p': "Prénom\_:"
'n': "Nom\_:"
```

**Comment by xabbuh** [comment]
Maybe I completely misunderstand what you try to do, but as far as I can see `\u00a0` is never interpreted as the non-breaking space by PHP. That's not related to what the Yaml component does (see also https://3v4l.org/b0vsK).

**Comment by Lozik** [comment]
Effectively! I did not know that. But I don't know where it comes from but if you put `\u00a0` in a yaml-based translation, it is interpreted identically to `\xc2\xa0` (so as non breakable space). It could be the Yaml parser, could be the translation component. (But it is definitely not twig, I checked that).
I don't know how to resolve this "issue" then - but I was certainly confused by having those differences and this inconsistency.
